src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java

```diff
@@ -326,15 +326,18 @@ protected boolean updateMedicalOutcomingQuantity(int medicalCode, double increme
 	 * @throws OHServiceException if an error occurs during the update.
 	 */
 	@SuppressWarnings("unchecked")
-	protected boolean updateMedicalWardQuantity(Ward ward, Medical medical, int quantity, Lot lot) throws OHServiceException {
+	protected boolean updateMedicalWardQuantity(Ward ward, Medical medical, int quantity, Lot lot)
+			throws OHServiceException {
 		MedicalWard medicalWard = medicalStockRepository.findOneWhereCodeAndMedicalAndLot(ward.getCode(), medical.getCode(), lot.getCode());
 
 		if (medicalWard != null) {
 			medicalWard.setIn_quantity(medicalWard.getIn_quantity() + quantity);
 			medicalStockRepository.save(medicalWard);
 		} else {
+			MedicalWard tmp = medicalStockRepository.findOneWhereCodeAndMedicalAndLotAndNotActive(ward.getCode(), medical.getCode(), lot.getCode());
+
 			medicalWard = new MedicalWard(ward, medical, quantity, 0, lot);
-			medicalStockRepository.insertMedicalWard(ward.getCode(), medical.getCode(), (double)quantity, lot.getCode());
+			medicalStockRepository.findAndUpdateOneWhereCodeAndMedicalAndLotAndNotActive(ward.getCode(), medical.getCode(), lot.getCode(), (double) quantity);
```

Comment:
why this method has changed?

Reply:
@AndreiDodu Up!

Reply:
if I am not wrong, I changed the name because the called method became an update, but if it is necessary, I will change it

Reply:
you are right, anyway this PR is closed, new PRs are

- https://github.com/informatici/openhospital-core/pull/818
- https://github.com/informatici/openhospital-gui/pull/1483

Any further development should then start from branches

- https://github.com/informatici/openhospital-core/tree/OP-780-oh-core-make-spring-use-xxx-active-fields-natively
- https://github.com/informatici/openhospital-gui/tree/OP-780-oh-core-make-spring-use-xxx-active-fields-natively